src/main/java/com/github/manolo8/darkbot/utils/DiscordWebhook.java

```diff
@@ -0,0 +1,258 @@
+package com.github.manolo8.darkbot.utils;
+
+import com.github.manolo8.darkbot.utils.http.Http;
+import com.github.manolo8.darkbot.utils.http.Method;
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+
+import java.awt.*;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Source: https://gist.github.com/k3kdude/fba6f6b37594eae3d6f9475330733bdb
+ */
+public class DiscordWebhook {
+    public static final Gson GSON = new Gson();
+    private final transient String url;
+    private final List<EmbedObject> embeds = new ArrayList<>();
+    private String content;
+    private String username;
+    @SerializedName("avatar_url") private String avatarUrl;
+    private boolean tts;
+
+    /**
+     * Constructs a new DiscordWebhook instance
+     *
+     * @param url The webhook URL obtained in Discord
+     */
+    public DiscordWebhook(String url) {
+        this.url = url;
+    }
+
+    public void setContent(String content) {
+        this.content = content;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public void setAvatarUrl(String avatarUrl) {
+        this.avatarUrl = avatarUrl;
+    }
+
+    public void setTts(boolean tts) {
+        this.tts = tts;
+    }
+
+    public void addEmbed(EmbedObject embed) {
+        this.embeds.add(embed);
+    }
+
+    public void execute() throws IOException {
+        if (this.content == null && this.embeds.isEmpty()) {
+            throw new IllegalArgumentException("Set content or add at least one EmbedObject");
+        }
+
+        Http http = Http.create(this.url, Method.POST);
+        HttpURLConnection conn = http.getConnection();
+        conn.addRequestProperty("Content-Type", "application/json");
+        conn.addRequestProperty("User-Agent", "Java-DiscordWebhook-BY-Gelox_");
+        conn.setDoOutput(true);
+
+        OutputStream stream = conn.getOutputStream();
+        stream.write(GSON.toJson(this).getBytes());
+        stream.flush();
+        stream.close();
+
+        http.closeInputStream();
+        conn.disconnect();
```

Comment:
This is quite a clusterfuck

To start with, Gson allows writing directly to an output stream, second of all, use try-with-resources instead of manually closing streams:

something like this:
```java
try (OutputStreamWriter writer = new OutputStreamWriter(http.getOutputStream()) {
    GSON.toJson(this, writer);
}
```